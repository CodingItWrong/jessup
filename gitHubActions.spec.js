const {getGitHubActionsConfig} = require('./gitHubActions');
const {FRAMEWORKS} = require('./script');

describe('getGitHubActionsConfig', () => {
  function getFramework(value) {
    return FRAMEWORKS.find(f => f.value === value);
  }

  describe('cra', () => {
    const framework = getFramework('cra');

    it('can generate a config without cypress', () => {
      expect(
        getGitHubActionsConfig({framework: 'cra', cypress: false}, framework)
      ).toMatchSnapshot();
    });

    it('can generate a config with cypress', () => {
      expect(
        getGitHubActionsConfig({framework: 'cra', cypress: true}, framework)
      ).toMatchSnapshot();
    });
  });

  describe('docusaurus', () => {
    const framework = getFramework('doc');

    it('can generate a config without cypress', () => {
      expect(
        getGitHubActionsConfig({framework: 'doc', cypress: false}, framework)
      ).toMatchSnapshot();
    });

    it('can generate a config with cypress', () => {
      expect(
        getGitHubActionsConfig({framework: 'doc', cypress: true}, framework)
      ).toMatchSnapshot();
    });
  });

  describe('expo', () => {
    const framework = getFramework('expo');

    it('can generate a config without any testing', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'expo', unitTesting: false, cypress: false},
          framework
        )
      ).toMatchSnapshot();
    });

    it('can generate a config with unit testing', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'expo', unitTesting: true, cypress: false},
          framework
        )
      ).toMatchSnapshot();
    });

    it('can generate a config with cypress', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'expo', unitTesting: false, cypress: true},
          framework
        )
      ).toMatchSnapshot();
    });

    it('can generate a config with both unit testing and cypress', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'expo', unitTesting: true, cypress: true},
          framework
        )
      ).toMatchSnapshot();
    });
  });

  describe('next', () => {
    const framework = getFramework('next');

    it('can generate a config without cypress', () => {
      expect(
        getGitHubActionsConfig({framework: 'next', cypress: false}, framework)
      ).toMatchSnapshot();
    });

    it('can generate a config with cypress', () => {
      expect(
        getGitHubActionsConfig({framework: 'next', cypress: true}, framework)
      ).toMatchSnapshot();
    });
  });

  describe('node', () => {
    const framework = getFramework('node');

    it('can generate a config without unit testing', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'node', unitTesting: false},
          framework
        )
      ).toMatchSnapshot();
    });

    it('can generate a config with unit testing', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'node', unitTesting: true},
          framework
        )
      ).toMatchSnapshot();
    });
  });

  describe('node with babel', () => {
    const framework = getFramework('babel');

    it('can generate a config without unit testing', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'babel', unitTesting: false},
          framework
        )
      ).toMatchSnapshot();
    });

    it('can generate a config with unit testing', () => {
      expect(
        getGitHubActionsConfig(
          {framework: 'babel', unitTesting: true},
          framework
        )
      ).toMatchSnapshot();
    });
  });

  describe('react native', () => {
    const framework = getFramework('rn');

    it('can generate a config without unit testing', () => {
      expect(
        getGitHubActionsConfig({framework: 'rn', unitTesting: false}, framework)
      ).toMatchSnapshot();
    });

    it('can generate a config with unit testing', () => {
      expect(
        getGitHubActionsConfig({framework: 'rn', unitTesting: true}, framework)
      ).toMatchSnapshot();
    });
  });
});
