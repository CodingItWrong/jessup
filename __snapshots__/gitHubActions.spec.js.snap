// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGitHubActionsConfig cra can generate a config with cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint

      - name: Cypress Tests
        uses: cypress-io/github-action@v4
        with:
          start: yarn start
          wait-on: 'http://localhost:3000'
"
`;

exports[`getGitHubActionsConfig cra can generate a config without cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig docusaurus can generate a config with cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint

      - name: Cypress Tests
        uses: cypress-io/github-action@v4
        with:
          start: yarn start
          wait-on: 'http://localhost:3000'
"
`;

exports[`getGitHubActionsConfig docusaurus can generate a config without cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig expo can generate a config with both unit testing and cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint

      - name: Install Expo CLI
        run: yarn global add expo-cli

      - name: Install Sharp CLI
        run: yarn global add sharp-cli

      - name: Cypress Tests
        uses: cypress-io/github-action@v4
        with:
          start: yarn web
          wait-on: 'http://localhost:19006'
"
`;

exports[`getGitHubActionsConfig expo can generate a config with cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint

      - name: Install Expo CLI
        run: yarn global add expo-cli

      - name: Install Sharp CLI
        run: yarn global add sharp-cli

      - name: Cypress Tests
        uses: cypress-io/github-action@v4
        with:
          start: yarn web
          wait-on: 'http://localhost:19006'
"
`;

exports[`getGitHubActionsConfig expo can generate a config with unit testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig expo can generate a config without any testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig next can generate a config with cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint

      - name: Cypress Tests
        uses: cypress-io/github-action@v4
        with:
          start: yarn dev
          wait-on: 'http://localhost:3000'
"
`;

exports[`getGitHubActionsConfig next can generate a config without cypress 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig node can generate a config with unit testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig node can generate a config without unit testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig node with babel can generate a config with unit testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig node with babel can generate a config without unit testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig react native can generate a config with e2e testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: macos-latest
    timeout-minutes: 60
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint

      - name: Cache Pods
        uses: actions/cache@v3
        id: podcache
        with:
          path: ios/Pods
          key: pods-\${{ hashFiles('**/Podfile.lock') }}

      - name: Update Pods
        if: steps.podcache.outputs.cache-hit != 'true'
        run: |
          gem update cocoapods xcodeproj
          cd ios && pod install && cd ..

      - run: brew tap wix/brew
      - run: brew install applesimutils
      - run: yarn detox build e2e --configuration ios.sim.release
      - run: yarn detox test e2e --configuration ios.sim.release --cleanup --debug-synchronization 200
"
`;

exports[`getGitHubActionsConfig react native can generate a config with unit and e2e testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: macos-latest
    timeout-minutes: 60
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint

      - name: Cache Pods
        uses: actions/cache@v3
        id: podcache
        with:
          path: ios/Pods
          key: pods-\${{ hashFiles('**/Podfile.lock') }}

      - name: Update Pods
        if: steps.podcache.outputs.cache-hit != 'true'
        run: |
          gem update cocoapods xcodeproj
          cd ios && pod install && cd ..

      - run: brew tap wix/brew
      - run: brew install applesimutils
      - run: yarn detox build e2e --configuration ios.sim.release
      - run: yarn detox test e2e --configuration ios.sim.release --cleanup --debug-synchronization 200
"
`;

exports[`getGitHubActionsConfig react native can generate a config with unit testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: Unit Tests
        run: yarn test --watchAll=false

      - name: ESLint
        run: yarn lint
"
`;

exports[`getGitHubActionsConfig react native can generate a config without any testing 1`] = `
"name: Test
on: [push]

jobs:
  test:
    name: Test
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v3

      - name: Get yarn cache directory path
        id: yarn-cache-dir-path
        run: echo \\"::set-output name=dir::$(yarn cache dir)\\"

      - uses: actions/cache@v3
        id: yarn-cache
        with:
          path: \${{ steps.yarn-cache-dir-path.outputs.dir }}
          key: \${{ runner.os }}-yarn-\${{ hashFiles('**/yarn.lock') }}
          restore-keys: |
            \${{ runner.os }}-yarn-

      - name: Install Dependencies
        run: yarn install --frozen-lockfile

      - name: ESLint
        run: yarn lint
"
`;
